src/processTargets/modifiers/scopeTypeStages/SubTokenStages.ts

```diff
@@ -0,0 +1,139 @@
+import { Range } from "vscode";
+import { SUBWORD_MATCHER } from "../../../core/constants";
+import { GRAPHEME_SPLIT_REGEX } from "../../../core/TokenGraphemeSplitter";
+import { Target } from "../../../typings/target.types";
+import {
+  ContainingScopeModifier,
+  EveryScopeModifier,
+} from "../../../typings/targetDescriptor.types";
+import { ProcessedTargetsContext } from "../../../typings/Types";
+import { matchAll } from "../../../util/regex";
+import { ModifierStage } from "../../PipelineStages.types";
+import { PlainTarget, SubTokenWordTarget } from "../../targets";
+import { getTokenRangeForSelection } from "./TokenStage";
+
+abstract class SubTokenStage implements ModifierStage {
+  constructor(
+    private modifier: ContainingScopeModifier | EveryScopeModifier,
+    private regex: RegExp
+  ) {}
+
+  run(context: ProcessedTargetsContext, target: Target): Target[] {
+    const tokenRange = getTokenRangeForSelection(
+      target.editor,
+      target.contentRange
+    );
+
+    const { document } = target.editor;
+    const text = document.getText(tokenRange);
+    const offset = document.offsetAt(tokenRange.start);
+
+    const contentRanges = matchAll<Range>(
+      text,
+      this.regex,
+      (match) =>
+        new Range(
+          document.positionAt(offset + match.index!),
+          document.positionAt(offset + match.index! + match[0].length)
+        )
+    );
+
+    const targets = this.createTargets(target, contentRanges);
+
+    if (this.modifier.type === "everyScope") {
+      return targets;
+    }
+
+    return [this.findSingleTarget(target, targets)];
+  }
+
+  private findSingleTarget(target: Target, targets: Target[]): Target {
+    let intersectingTargets = targets
+      .map((t) => ({
+        target: t,
+        intersection: t.contentRange.intersection(target.contentRange),
+      }))
+      .filter((it) => it.intersection != null);
+
+    // Empty range utilize single leftmost target
+    if (target.contentRange.isEmpty) {
+      return intersectingTargets[0].target;
+    }
+
+    // On non empty range utilize all non-empty intersecting targets
+    intersectingTargets = intersectingTargets.filter(
+      (it) => !it.intersection!.isEmpty
+    );
```

Comment:
is this what we do for eg "token", "block" and "line"?  I'd be tempted to expand on adjacent but idk.  Maybe worth discussing at meet-up

I see now why you're doing the `map` above.  Might be able to remove it depending what we decide here

Reply:
That would mean that if you have a single character selected we would expand to all three of them. Probably not to the desired behavior